packages/datagateway-download/src/downloadApiHooks.ts

```diff
@@ -250,3 +327,434 @@ export const useDatafileCounts = (
 
   return queries;
 };
+
+export interface UseDownloadParams {
+  id: number;
+}
+
+/**
+ * A React hook that fetches a single download with the given id.
+ * useQuery options can be passed in, which will override the default used.
+ *
+ * Example:
+ * ```
+ * useDownload({
+ *   id: 123,
+ *   select: (download) => format(download)
+ * })
+ * ```
+ */
+export const useDownload = <T = Download>({
+  id,
+  ...queryOptions
+}: UseDownloadParams &
+  UseQueryOptions<
+    Download,
+    AxiosError,
+    T,
+    (number | QueryKey)[]
+  >): UseQueryResult<T, AxiosError> => {
+  // Load the download settings for use.
+  const downloadSettings = React.useContext(DownloadSettingsContext);
+
+  return useQuery(
+    [QueryKey.DOWNLOAD, id],
+    () =>
+      getDownload(id, {
+        facilityName: downloadSettings.facilityName,
+        downloadApiUrl: downloadSettings.downloadApiUrl,
+      }),
+    {
+      onError: (error) => {
+        handleICATError(error);
+      },
+      retry: retryICATErrors,
+      ...queryOptions,
+    }
+  );
+};
+
+/**
+ * A React hook that fetches all downloads created by the user.
+ */
+export const useDownloads = (): UseQueryResult<
+  FormattedDownload[],
+  AxiosError
+> => {
+  // Load the download settings for use.
+  const downloadSettings = React.useContext(DownloadSettingsContext);
+  const { formatDownload } = useDownloadFormatter();
+
+  return useQuery(
+    QueryKey.DOWNLOADS,
+    () =>
+      fetchDownloads({
+        facilityName: downloadSettings.facilityName,
+        downloadApiUrl: downloadSettings.downloadApiUrl,
+      }),
+    {
+      select: (downloads) => downloads.map(formatDownload),
+      onError: (error) => {
+        handleICATError(error);
+      },
+      retry: retryICATErrors,
+    }
+  );
+};
+
+export interface UseDownloadDeletedParams {
+  downloadId: number;
+  deleted: boolean;
+}
+
+/**
+ * A React query that provides a mutation for deleting a download item.
+ */
+export const useDownloadDeleted = (): UseMutationResult<
```

Comment:
Minor comment - but in my head it makes more sense to name mutations after action they perform i.e. this would be `useDeleteDownload` as it implies that this hook can be used to delete downloads.

Reply:
It was actually named `useDeleteDownload` originally, but then I realized it could also be used to restore a download, so I renamed it to `useDownloadDeleted`. Maybe `useDeleteOrRestoreDownload`?